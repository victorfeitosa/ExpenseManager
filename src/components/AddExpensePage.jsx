import React from 'react';
import { connect } from 'react-redux';
import { addExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

const AddExpensePage = (props) => (
  <div>
    <h1>Add expenses</h1>
    <ExpenseForm
      buttonText="Add expense"
      onSubmit={(expense) => {
        props.dispatch(addExpense(expense));
        props.history.push('/');
      }} />
  </div>
);

export default connect()(AddExpensePage);