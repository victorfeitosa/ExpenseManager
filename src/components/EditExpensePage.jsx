import React from 'react';
import { connect } from 'react-redux';
import { editExpense, removeExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

const EditExpensePage = (props) => {
  return (
    <div>
      <ExpenseForm
        expense={props.expense}
        buttonText="Edit expense"
        onSubmit={(expense) => {
          console.log(expense);
          props.dispatch(editExpense({
            id: props.expense.id,
            updates: expense
          }));
          props.history.push('/');
        }}>
      </ExpenseForm>
      <button onClick={() => {
        props.dispatch(removeExpense({ id: props.expense.id }));
        props.history.push('/');
      }}>Remove Expense &times;</button>
    </div>
  );
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((e) => e.id === props.match.params.id)
  }
};

export default connect(mapStateToProps)(EditExpensePage);
