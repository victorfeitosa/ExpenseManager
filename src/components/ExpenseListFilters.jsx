import React from 'react';
import { DateRangePicker } from 'react-dates';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';

class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  };

  onDatesChange = ({startDate, endDate}) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  }

  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  }

  render () {
    return (
      <div>
        <input type="text" value={this.props.filters.text} onChange={(e) => {
          this.props.dispatch(setTextFilter(e.target.value));
        }}/>
        <select
          value={this.props.filters.sortBy}
          onChange={(e) => {
            if (e.target.value === 'date') this.props.dispatch(sortByDate());
            else this.props.dispatch(sortByAmount());
          }}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          startDateId="filter_start_date"
          endDate={this.props.filters.endDate}
          endDateId="filter_end_date"
          focusedInput={this.state.calendarFocused}
          numberOfMonths={1}
          isOutsideRange={() => false}
          showClearDates={true}
          onDatesChange={this.onDatesChange}
          onFocusChange={this.onFocusChange}
        >
        </DateRangePicker>
      </div>);
  }
}

const mapStateToProps = (state) => ({
  filters: state.filters
});

export default connect(mapStateToProps)(ExpenseListFilters);
