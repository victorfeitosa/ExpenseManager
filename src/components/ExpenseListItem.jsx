import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({id, description, amount, createdAt}) => (
  <div>
    <p>{description}</p>
    <b>{amount}</b>
    <p>{createdAt}</p>

    <Link to={'/edit/' + id.toString()}>Edit</Link>
  </div>
);

export default ExpenseListItem;
