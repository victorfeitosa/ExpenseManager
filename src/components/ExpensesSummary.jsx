import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expensesTotal';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
   const expensesWord = expenseCount === 1 ? 'expense' : 'expenses';
   return (
      <div className='page-header'>
         <div className='content-container'>
            <h1 className='page-header__title'>Viewing <span>{expenseCount}</span> {expensesWord} totalling <span>{numeral(expensesTotal / 100).format('$0,0.00')}</span></h1>
            <div className='page-header__actions'>
               <Link className='button' to='/create'>Add expense</Link>
            </div>
         </div>
      </div>
   )
}

const mapStateToProps = (state) => {
   const visibleExpenses = selectExpenses(state.expenses, state.filters);
   return {
      expenseCount: visibleExpenses.length,
      expensesTotal: expensesTotal(visibleExpenses)
   }
};

export default connect(mapStateToProps)(ExpensesSummary);
