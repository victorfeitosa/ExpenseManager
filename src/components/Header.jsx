import React from "react";
import { connect } from "react-redux";
import { NavLink } from "react-router-dom";
import { startLogout } from "../actions/auth";

export const Header = ({ startLogout }) => (
  <header>
    {true && (
      <div>
        <h1>Expense Manager</h1>
        <nav>
          <NavLink to="/" activeClassName="is-active" exact={true}>
            Home
          </NavLink>
          <NavLink to="/create" activeClassName="is-active">
            Create expense
          </NavLink>
          <NavLink to="/help" activeClassName="is-active">
            Help
          </NavLink>
          <button onClick={startLogout}>Logout</button>
        </nav>
      </div>
    )}
  </header>
);

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(
  null,
  mapDispatchToProps
)(Header);
