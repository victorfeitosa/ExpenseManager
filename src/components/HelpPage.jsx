import React from 'react';

const HelpPage = () => (
   <p>HELP</p>
);
export default HelpPage;
