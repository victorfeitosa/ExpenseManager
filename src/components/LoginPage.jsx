import React, { Component } from 'react'
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
  <div className='box-layout'>
    <div className='box-layout__box'>
      <h1 className='box-layout__title'>Expense Manager</h1>
      <p>Have your expenses under control.</p>
      <button className='button' onClick={startLogin}>Login witgh Google</button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
