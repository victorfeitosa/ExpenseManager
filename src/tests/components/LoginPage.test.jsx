import React from "react";
import { shallow } from "enzyme";
import { LoginPage } from "../../components/LoginPage";

let wrapper;

beforeEach(() => {
  wrapper = shallow(<LoginPage />);
});

test("should render LoginPage correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should call startLogin on button click", () => {
  const startLogin = jest.fn();
  const wrapper = shallow(<LoginPage loginWithGithub={startLogin} loginWithGoogle={startLogin}/>);
  wrapper.find("#login-google").simulate("click");
  wrapper.find("#login-github").simulate("click");
  expect(startLogin).toHaveBeenCalledTimes(2);
});
