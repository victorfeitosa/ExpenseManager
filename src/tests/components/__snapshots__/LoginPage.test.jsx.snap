// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="box-layout__box"
      >
        <h1
          className="box-layout__title"
        >
          Expense Manager
        </h1>
        <p>
          Have your expenses under control.
        </p>
        <h3>
          Login with
        </h3>
        <button
          className="button button--wide button--secondary"
          id="login-google"
          title="Google"
        >
          <FaGoogle />
        </button>
        <button
          className="button button--wide button--tertiary"
          id="login-github"
          title="GitHub"
        >
          <FaGithub />
        </button>
      </div>,
      "className": "box-layout",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="box-layout__title"
          >
            Expense Manager
          </h1>,
          <p>
            Have your expenses under control.
          </p>,
          <h3>
            Login with
          </h3>,
          <button
            className="button button--wide button--secondary"
            id="login-google"
            title="Google"
          >
            <FaGoogle />
          </button>,
          <button
            className="button button--wide button--tertiary"
            id="login-github"
            title="GitHub"
          >
            <FaGithub />
          </button>,
        ],
        "className": "box-layout__box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Expense Manager",
            "className": "box-layout__title",
          },
          "ref": null,
          "rendered": "Expense Manager",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Have your expenses under control.",
          },
          "ref": null,
          "rendered": "Have your expenses under control.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login with",
          },
          "ref": null,
          "rendered": "Login with",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FaGoogle />,
            "className": "button button--wide button--secondary",
            "id": "login-google",
            "onClick": undefined,
            "title": "Google",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FaGithub />,
            "className": "button button--wide button--tertiary",
            "id": "login-github",
            "onClick": undefined,
            "title": "GitHub",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="box-layout__box"
        >
          <h1
            className="box-layout__title"
          >
            Expense Manager
          </h1>
          <p>
            Have your expenses under control.
          </p>
          <h3>
            Login with
          </h3>
          <button
            className="button button--wide button--secondary"
            id="login-google"
            title="Google"
          >
            <FaGoogle />
          </button>
          <button
            className="button button--wide button--tertiary"
            id="login-github"
            title="GitHub"
          >
            <FaGithub />
          </button>
        </div>,
        "className": "box-layout",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="box-layout__title"
            >
              Expense Manager
            </h1>,
            <p>
              Have your expenses under control.
            </p>,
            <h3>
              Login with
            </h3>,
            <button
              className="button button--wide button--secondary"
              id="login-google"
              title="Google"
            >
              <FaGoogle />
            </button>,
            <button
              className="button button--wide button--tertiary"
              id="login-github"
              title="GitHub"
            >
              <FaGithub />
            </button>,
          ],
          "className": "box-layout__box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Expense Manager",
              "className": "box-layout__title",
            },
            "ref": null,
            "rendered": "Expense Manager",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Have your expenses under control.",
            },
            "ref": null,
            "rendered": "Have your expenses under control.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login with",
            },
            "ref": null,
            "rendered": "Login with",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FaGoogle />,
              "className": "button button--wide button--secondary",
              "id": "login-google",
              "onClick": undefined,
              "title": "Google",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FaGithub />,
              "className": "button button--wide button--tertiary",
              "id": "login-github",
              "onClick": undefined,
              "title": "GitHub",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
